archive-tar.c

```diff
@@ -14,9 +14,8 @@
 
```

Comment:
The second paragraph of the commit message talks about "our implementation", but it actually wants to talk about the next commit (because there is no "our implementation" before the next commit yet...).

Reply:
Except for the line "replace write_or_die() calls with write_block_or_die()", this commit message would now fit better onto the second commit than onto this one...

Reply:
A better commit message for *this here* commit might be:

```
archive: replace write_or_die() calls with write_block_or_die()

In the next commit, we will change the gzip compression so
that we no longer spawn `gzip` but let zlib perform the
compression in the same process.

In preparation for this, we consolidate all the block writes into a
single function.
```